### ISSUE TITLE
Empty NBT tag causing items to not stack

### ISSUE THREAD

**Issue by VeneratedLerkur** [dkmk100/ArsOmega#77]
I'm on arsomega-1.19.2-3.1.1.jar and seem to have encountered a bug that was listed as closed. I was having the issues of empty NBT tags becoming applied to items I would break and pick up. Which makes them not stack. I tried updating to the beta version of 1.19.2 and still seemed to have the issue. Was having a hard time troubleshooting and was wondering if anyone else would be able to replicate this? Once I removed the mod, everything seemed to stack again.

**Comment by dkmk100** [comment]
Do you have any idea what you were doing when these items were created?
Also, on the new beta, do new items get this issue or is it just that old ones keep the empty tag? I can't remove the tag from items but I fixed the old cause of the issue so if they still get created there must be another version of the bug in the mod

**Comment by VeneratedLerkur** [comment]
tldr; rant about how it might have something to do with breaking/dropping or picking up blocks.

so I'm not sure exactly what I'm doing that's causing it but at least maybe an idea for you...

I'm playing on a server I made for me and some friends. We discovered that some things don't stack for some reason but if you drop them onto the ground and then pick them up they will stack. sometimes you have to drop both items to get them to stack(as in sometimes I can drop 1 item and pick it up and it goes to the stack, and sometimes I must drop both). This was only a minor nuisance at the time until I started using build wands. Things that have the tag won't be registered by the build wand as a resource to pull from. My frustration and couple hours of thread hunting lead me to discovering the tagged blocks, and then I found a small thread that pointed me towards ars omega.

It seems like its tagging the blocks that I pick up intermittently. Also The tagged blocks seem to have priority when doing the, lets say, "drop to stack" tactic. And one more thing to possibly note is that I can have untagged blocks in my inventory (freshly crafted), build with the build wand, and then break and pick up some items and some of them won't stack.
I'll keep looking as I have the time and I'll let my friends know too so they can try to look for specifics.

**Comment by dkmk100** [comment]
Does it happen in singleplayer, or only on the server?
What happens to the items that don't stack if you disconnect and join back on the server?
Looking at the code I don't see anything to suggest I could be adding empty tags to items... 
Which modpack are you using? 

Are you able to make a backup of the world and check whether removing Omega actually stops this from happening?

**Comment by VeneratedLerkur** [comment]
I'm on a server, I actually don't play single player worlds because I always have a problem with my own single player worlds.

Relogging the game does nothing to the stacks.

The modpack is one I made called "an old fable" 

Something I did:
1. Made server backup
2. Had 32 tagged and 32 untagged blocks
3. Updated to beta.
4. The two half stacks still didn't stack
5. Removed omega entirely
6. Logged in and blocks stacked
7. Restored backup because I still want the awesome mod xD

**Comment by dkmk100** [comment]
wait, the blocks stacked after removing Omega? 
can I have the latest.log from the server when you're playing with omega?

**Comment by dkmk100** [comment]
also, has this been going on since you created the world or is it only recently?
What are you using to break the blocks?

**Comment by VeneratedLerkur** [comment]
> wait, the blocks stacked after removing Omega? can I have the latest.log from the server when you're playing with omega?

Unless I made some kind of mistake while testing the hypothesis which is entirely possible, yes. 

> also, has this been going on since you created the world or is it only recently? What are you using to break the blocks?

Since creation. I am using an emerald pickaxe from easy emerald, and a projectile break spell from Mana and Artifice. I'm also wondering if maybe its blocks found in loot chests that spawn with omega items(mods like evilcraft that seem to mix all the loot tables maybe) and then they are getting mixed with the blocks that I have, and I just didn't notice that.

[latest.log](https://github.com/dkmk100/ArsOmega/files/14817792/latest.log)
I looked through the latest.log and I it was mostly just things like "There are 1 of a max of 20 players online"

man I really hope that I'm not being insane and making you go on a wild goose chase. I'll be able to do more testing tomorrow.

**Comment by VeneratedLerkur** [comment]
I just caught it happening! I was in a cave and broke a jar, it dropped torches but my inventory was full even though I had torches on my hotbar. so I ran the command first before i picked up the new torches. the torches in my inv didn't have the tag{}}. However, after I made space in my inventory the new torches dropped from the jar DID have the tag{}}

actually now that im looking closer... everything i mine and pick up is becoming tagged...

**Comment by dkmk100** [comment]
I downloaded the pack, deleted Omega, and was still able to replicate the bug with a torch. Thus, I'm pretty confident this isn't related to Omega. 

Steps to confirm this yourself:
-remove Omega and load a copy of the world
-craft or otherwise obtain brand new torches
-place and break one of them
-it won't stack as it has an empty tag

**Comment by VeneratedLerkur** [comment]
aw shoot. sorry!
